src/setup.py

```diff
@@ -8,7 +8,14 @@
         "pdist_agregation",
         sources=["_pdist_agregation.pyx"],
         include_dirs=[numpy.get_include()],
-        extra_compile_args=['-fopenmp'],
+        extra_compile_args=[
+            '-fopenmp',
+            '-O3',
+            '-ftree-vectorize',
+            '-march=native',
```

Comment:
For reference this option does not exist when compiling with clang-11 apparently.

Reply:
I've managed to compile with clang 12, but it's probably better reverting `march` and `mtune` for portability, yes.